packages/webpack/src/config/base.js

```diff
@@ -10,14 +10,16 @@ import TerserWebpackPlugin from 'terser-webpack-plugin'
 import WebpackBar from 'webpackbar'
 import env from 'std-env'
 
-import { isUrl, urlJoin } from '@nuxt/utils'
+import { createTimeHash, isUrl, urlJoin } from '@nuxt/utils'
 
 import PerfLoader from '../utils/perf-loader'
 import StyleLoader from '../utils/style-loader'
 import WarningIgnorePlugin from '../plugins/warning-ignore'
 
 import { reservedVueTags } from '../utils/reserved-tags'
 
+const BUILD_HASH = createTimeHash()
```

Comment:
I think this would be better to at least set inside constructor.

Reply:
won't the hash be different for server and client then?

Reply:
Builder instance is shared.

Reply:
Ah, so you want to add it in the Builder itself instead of the webpack config?

Reply:
Yep :) Inside build class constructor. So we can also use it by generator.

Reply:
@pi0 Sure, we can do that. But this still wouldn't solve the `nuxt generate --no-build` use case, right?

Reply:
Yeah. Unless storing in a manifest file after build and restore on generate --no-build

Reply:
That'd work in terms of sharing, but then the hashes wouldn't be different  (and we'd have to keep relying on the export hash)

Reply:
Yep. --no-build is not relative to this change request. Maybe we should use a fresh hash on each generate.